src/BoilerPlate/App.cpp

```diff
@@ -101,8 +129,39 @@ namespace Engine
 			SDL_Log("Right key was pressed.");
 			m_player->RotateRight();
 			break;
+
+		case SDL_SCANCODE_D: 
+		{
+			SDL_Log("D key was pressed.");
+			m_player->ShowBoundingCircles(true);
```

Comment:
I'd make this a mode that the entities query before rendering. This way you don't have to call the `ShowBoundingCircles` function every time. Maybe entities have a parent class? It could be called `Game`. This observation is more of an optimization, rather than something necessary for the assignment.